data/src/main/java/utill/DMAPIhandler.kt

```
@@ -0,0 +1,66 @@
+class DMAPIhandIer<T> {
+    private lateinit var httpRequest: suspend () -> T
+    fun httpRequest(httpRequest: suspend () -> T) =
+        this.apply { this.httpRequest = httpRequest }
+
+    suspend fun sendRequest(): T {
+        return try {
+            Log.d("ApiHandler", "Success")
+            withContext(Dispatchers.IO) {
+                httpRequest.invoke()
+            }
+        } catch (e: HttpException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)